src/models/relations.js

```diff
@@ -316,13 +316,19 @@ export default class Relations extends EventEmitter {
             return event;
         }, null);
     }
+
     /*
-     * @param {MatrixEvent} targetEvent the event related to.
+     * @param {MatrixEvent} targetEvent the event the relations are related to.
      */
     setTargetEvent(event) {
-        this._targetEvent = event;
-        if (this.relationType === "m.replace") {
-            this._targetEvent.makeReplaced(this.getLastReplacement());
+        if (!this._targetEvent) {
```

Comment:
Maybe do:

```
if (this._targetEvent) {
    return;
}
```

for less indenting? (I am a bit "early return obsessive", feel free to ignore...)